examples/speech_synthesis/incremental_text_to_speech/generate_waveform_incremental.py

```diff
@@ -0,0 +1,253 @@
+                        help="Use pseudo lookahead from ST system")
+    parser.add_argument("--generate-audio-with-discontinuity", action="store_true",
+                        help="If true, include silence (due to no output playing) in the synthesized utterances")
+    return parser
+
+
+def main(args):
+    assert (args.dump_features or args.dump_waveforms or args.dump_attentions
+            or args.dump_eos_probs or args.dump_plots)
+
+    if args.generate_audio_with_discontinuity:
+        assert args.incremental_tts_input_from_simuleval, \
+            "Output with discontinuity is only possible when input is timestamped (by simuleval log file)."
+
+    if args.use_pseudo_lookahead:
+        assert args.lookahead_words == 0, "Pseudo lookahead only works when not using actual lookahead."
+
+    if args.max_tokens is None and args.batch_size is None:
+        args.max_tokens = 8000
+    logger.info(args)
+
+    use_cuda = torch.cuda.is_available() and not args.cpu
+    task = tasks.setup_task(args)
+    models, saved_cfg, task = checkpoint_utils.load_model_ensemble_and_task(
+        [args.path],
+        task=task,
+    )
+    model = models[0].cuda() if use_cuda else models[0]
+    task.args.n_frames_per_step = saved_cfg.task.n_frames_per_step
+    task.load_dataset(args.gen_subset, task_cfg=saved_cfg.task)
+
+    data_cfg = task.data_cfg
+    sample_rate = data_cfg.config.get("features", {}).get("sample_rate", 22050)
+    resample_fn = {
+        False: lambda x: x,
+        True: lambda x: torchaudio.sox_effects.apply_effects_tensor(
+            x.detach().cpu().unsqueeze(0), sample_rate,
+            [['rate', str(args.output_sample_rate)]]
+        )[0].squeeze(0)
+    }.get(args.output_sample_rate != sample_rate)
+    if args.output_sample_rate != sample_rate:
+        logger.info(f"resampling to {args.output_sample_rate}Hz")
+
+    generator = task.build_generator([model], args)
+    Path(args.results_path).mkdir(exist_ok=True, parents=True)
+    is_na_model = getattr(model, "NON_AUTOREGRESSIVE", False)
+    vocoder = task.args.vocoder
+
+    device = torch.device('cuda') if use_cuda else torch.device('cpu')
+    main_results_path = Path(args.results_path)
+
+    if args.incremental_tts_input_from_simuleval:
+        logger.info(f"Generating waveforms from SimulEval log file {args.incremental_tts_input_from_simuleval}")
+        with open(args.incremental_tts_input_text, "r") as f:
+            lines = f.readlines()
+    else:
+        logger.info(f"Generating waveforms from txt file {args.incremental_tts_input_from_simuleval}")
+        lines = []
+        with open(args.incremental_tts_input_text, "r") as f:
+            for l in f.readlines():
+                lines.append(l.strip())
+    phonemizer = Phonemizer(args.phonemizer_lang)
+
+    with progress_bar.build_progress_bar(args, enumerate(lines)) as t:
+        for s in t:
+            sample_idx, entry = s
+
+            if sample_idx < 3600:
```

Comment:
use constant to explain this number

Reply:
Sorry, that was a line left over from test runs. :sweat_smile: I've removed it.